src/binding.ts

```diff
@@ -4,61 +4,195 @@
  * sonarlint@sonarsource.com
  * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
-'use strict';
 
 import * as VSCode from 'vscode';
+import { writeFileSync, mkdirSync, existsSync } from 'fs';
+import { Commands } from './commands';
+import { ConnectionSettingsService } from './settings';
 
 const SONARLINT_CATEGORY = 'sonarlint';
-const PROJECT_BINDING_PROPERTY = 'connectedMode.project';
+const BINDING_SETTINGS = 'connectedMode.project';
+const DEFAULT_CONNECTION_ID = '<default>';
+const OPEN_FOLDER_ACTION = 'Open Folder';
+const BIND_MANUALLY_ACTION = 'Bind Manually';
+
+export interface ProjectBinding {
+    connectionId?: string;
```

Comment:
Indentation is 4-spaces instead of 2 (also in the test file) - maybe there is something missing in your IDE configuration?